neqo-transport/src/connection/mod.rs

```diff
@@ -1531,7 +1531,8 @@ impl Connection {
                 if let CloseError::Transport(_) = error_code {
                     builder.encode_varint(*frame_type);
                 }
-                builder.encode_vvec(reason_phrase);
+                let reason_len = min(min(reason_phrase.len(), 256), builder.remaining());
+                builder.encode_vvec(&reason_phrase[..reason_len]);
```

Comment:
This is going to overflow the remaining space as a result of the length field.  As the length is limited to 256 here, that means you might need two bytes for that length.

```suggestion
                let reason_len = min(min(reason_phrase.len(), 256), builder.remaining() - 2);
                builder.encode_vvec(&reason_phrase[..reason_len]);
```